libs/types/src/tokens.rs

```diff
@@ -255,6 +259,64 @@ impl CrossChainTransferability {
 	}
 }
 
+/// Connectors wrapped tokens
+///
+/// Currently, Connectors are only deployed on EVM-based chains and therefore
+/// the only supported type of connectors-wrapped tokens are Evm tokens. In the
+/// far future, we might support wrapped tokens from other chains, such as
+/// Cosmos, Avalanche, etc.
+#[derive(
+	Clone, Copy, PartialOrd, Ord, PartialEq, Eq, Debug, Encode, Decode, TypeInfo, MaxEncodedLen,
+)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub enum ConnectorsWrappedToken {
+	/// An EVM-native token
+	EVM {
+		/// The EVM chain id where the token is deployed
+		chain_id: EVMChainId,
+		/// The token contract address
+		address: [u8; 20],
+	},
+}
+
+impl TryFrom<MultiLocation> for ConnectorsWrappedToken {
+	type Error = ();
+
+	fn try_from(location: MultiLocation) -> Result<Self, Self::Error> {
+		match location {
+			MultiLocation {
+				parents: 0,
+				interior:
+					X2(
+						GlobalConsensus(NetworkId::Ethereum { chain_id }),
+						AccountKey20 {
+							network: None,
+							key: address,
+						},
+					),
+			} => Ok(Self::EVM { chain_id, address }),
+			_ => Err(()),
+		}
+	}
+}
+
+impl Into<MultiLocation> for ConnectorsWrappedToken {
```

Comment:
This basically defines how we describe an EVM Connectors wrapped token as a `MultiLocation`.
The non-final structure proposed here is to have `parents: 0` to symbolise that it's relative to the local chain, and then in the `interior` we say that it's part of the  `GlobalConsensus(NetworkId::Ethereum { chain_id }),`, and finally the `AccountKey20` with the evm token address.

The way I read this is "a Centrifuge-wrapped token that is actually an EVM token". Now, what I am missing here is  being more specific that this is a Centrifuge **Connectors**-wrapped token.

We could address that by making the `interior` an `X3 ( PalletInstance( ConnectorsPalltetIndex), GlobalConsensus(NetworkId::Ethereum { chain_id }), AccountKey20  { .. }`. This would mean that conversion `ConnectorsWrappedToken` to `MultiLocation` becomes runtime-dependant, but that would be fine.

Thoughts?

Reply:
> This would mean that conversion ConnectorsWrappedToken to MultiLocation becomes runtime-dependant, but that would be fine.

IMO, that is totally fine and I favour this variant over the `X2` one.

Reply:
Yes, me too. I started with X2 to kick off this discussion with something to look at already and because I recall @mustermeiszer not being into the idea of including the connectors pallet index here when I suggest this a while back 🤔

Reply:
@wischli @mustermeiszer I have implemented the alternative containing the X3 with the Connectors pallet index. I have done so following a similar pattern used with the `CurrencyIdConvert` type. This is necessary given that the conversion to make it generic on the specific runtime it will be used while being runtime-dependant, and since adding the pallet index to the `ConnectorsWrappedToken` type didn't make sense.

Reply:
My two cents are, that we never leak this `ConnectorsWrapedToken` to the outside, right? It is only used as we need a way to verify the arguements of the users, so making this runtime dependent seems to be leaking a type that has no use to the outside. But maybe I am missing something.

Reply:
There's for sure validity in that argument. But I do find that being specific in the structure of a MultiLocation is beneficial because it makes it more future-proof; e.g. imagine we would have some new feature in the future involving wrapped tokens with a similar nature. Not being able to distinguish between them looking at the MultiLocation is a drawback if not a problem, it would require migrations, etc. So if we can compromise on the current solution I'd vote to keep it 👍